datasize/src/lib.rs

```diff
@@ -349,6 +349,25 @@ where
     }
 }
 
+impl<T, E> DataSize for Result<T, E>
+where
+    T: DataSize,
+    E: DataSize,
+{
+    // Results are only not dynamic if their types have no heap data at all and are not dynamic.
+    const IS_DYNAMIC: bool =
+        (T::IS_DYNAMIC || E::IS_DYNAMIC || T::STATIC_HEAP_SIZE > 0 || E::STATIC_HEAP_SIZE > 0);
+
+    const STATIC_HEAP_SIZE: usize = 0;
```

Comment:
This would be something like `if T::STATIC_HEAP_SIZE == E::STATIC_HEAP_SIZE { T::STATIC_HEAP_SIZE } else { 0 }` I believe. However it cannot be written that way (no way to use `if` in `const` eval.

Reply:
So the workaround is likely something like

```rust
const STATIC_HEAP_SIZE: usize = [0usize, T::STATIC_HEAP_SIZE][(T::STATIC_HEAP_SIZE == E::STATIC_HEAP_SIZE) as usize];
```

Reply:
Wouldn't it be more like `min(T::STATIC_HEAP_SIZE, E::STATIC_HEAP_SIZE)`?

Reply:
You're right. I gotchu, fam: https://github.com/CasperLabs/datasize-rs/blob/922c65ebe4b1d5b26a6ad272944ecf81ba53f652/datasize_derive/src/lib.rs#L379-L385

Reply:
You can move that function out of the comment and into the actual `datasize-rs` and publish it.